components/scream/cmake/ScreamMacros.cmake

```diff
@@ -0,0 +1,31 @@
+# This macro takes the following arguments:
+#    - target_name: the name of the executable
+#    - target_srcs: a list of src files for the executable
+#      Note: no need to include catch_main; this macro will add it
+#    - scream_libs: a list of scream libraries needed by the executable
+#    - config_defines: a list of additional defines for the compiler
+#    - num_mpi_ranks: the number of mpi ranks for the test
+#    - config_defines: specific define directives to be passed to the compiler
+
+MACRO(CreateUnitTest target_name target_srcs scream_libs num_mpi_ranks config_defines)
```

Comment:
Is there a reason this is a macro? Functions are nicer since they don't pollute the namespace.

Reply:
That's true. In my first implementation I didn't pass all the args in the signature (the macro _assumed_ they were defined). Then I didn't like that assumption, so I passed all of them. So now I can simply turn it into a function.